src/blob/index/core.rs

```diff
@@ -305,19 +317,19 @@ where
 
     async fn get_any(&self, key: &K) -> Result<Option<RecordHeader>> {
         debug!("index get any");
-        match &self.inner {
+        let result = match &self.inner {
             State::InMemory(headers) => {
                 debug!("index get any in memory headers: {}", headers.len());
                 // in memory indexes with same key are stored in ascending order, so the last
                 // by adding time record is last in list (in b+tree disk index it's first)
-                Ok(headers.get(key).and_then(|h| h.last()).cloned())
+                headers.get(key).and_then(|h| h.last()).cloned()
             }
             State::OnDisk(findex) => {
                 debug!("index get any on disk");
-                let header = findex.get_any(key).await?;
-                Ok(header)
+                findex.get_any(key).await?
             }
-        }
+        };
+        Ok(result.and_then(|h| if h.is_deleted() { None } else { Some(h) }))
```

Comment:
Function `get_all` above should also be updated in the same way. Please, check other functions: maybe they also should be updated